src/main/java/org/graylog/integrations/aws/service/KinesisService.java

```diff
@@ -380,4 +387,42 @@ Record selectRandomRecord(List<Record> recordsList) {
         LOG.debug("Selecting a random Record from the sample list.");
         return recordsList.get(new Random().nextInt(recordsList.size()));
     }
+
+    /**
+     * Creates a new Kinesis stream.
+     *
+     * @param kinesisNewStreamRequest request which contains region, access, secret, region, streamName and shardCount
+     * @return the status response
+     */
+    public KinesisNewStreamResponse createNewKinesisStream(KinesisNewStreamRequest kinesisNewStreamRequest) {
+        LOG.debug("Creating Kinesis client with the provided credentials.");
+        KinesisClient kinesisClient = createClient(kinesisNewStreamRequest.region(),
+                                                   kinesisNewStreamRequest.awsAccessKeyId(),
+                                                   kinesisNewStreamRequest.awsSecretAccessKey());
+
+        LOG.debug("Creating new Kinesis stream request [{}].", kinesisNewStreamRequest.streamName());
+        CreateStreamRequest createStreamRequest = CreateStreamRequest.builder()
+                                                                     .streamName(kinesisNewStreamRequest.streamName())
+                                                                     .shardCount(SHARD_COUNT)
+                                                                     .build();
+        LOG.debug("Sending request to create new Kinesis stream [{}] with [{}] shards.",
+                  kinesisNewStreamRequest.streamName(), SHARD_COUNT);
+
+        String responseMessage;
+        try {
+            kinesisClient.createStream(createStreamRequest);
+            responseMessage = String.format("Success. The new stream [%s] was created with [%d] shards.",
+                                            kinesisNewStreamRequest.streamName(), SHARD_COUNT);
+            return KinesisNewStreamResponse.create(responseMessage, new HashMap<>());
+        } catch (Exception e) {
+
+            String specificError = ExceptionUtils.formatMessageCause(e);
+            responseMessage = String.format("Attempt to create [%s] new Kinesis stream" +
```

Comment:
Need an extra space after "stream", since the error has reads like this with a missing space: "Attempt to create [new-dan-test] new Kinesis streamwith [1] shards failed"
```suggestion
            responseMessage = String.format("Attempt to create [%s] new Kinesis stream " +
```